tests/src/Interop/SizeConst/SizeConstTest.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+
+[StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi)]
+public struct S_CHARArray_ByValTStr
+{
+    [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 4)]
+    public string arr;
+    public S_CHARArray_ByValTStr(string parr) { arr = parr; }
+}
+
+class Test
+{
+    internal const int ARRAY_SIZE = 100;
+
+    //UnmanagedType.ByValTStr
+    [DllImport("SizeConstNative",CharSet = CharSet.Ansi)]
+    static extern bool TakeByValTStr(S_CHARArray_ByValTStr s, int size);
+
+    static bool SizeConstByValTStr()
+    {
+        // always marshal managedArray.Length
+        S_CHARArray_ByValTStr s = new S_CHARArray_ByValTStr();
+        s.arr = "有个可爱";
```

Comment:
Yea , I know that .. I don't see a way to test this without changing machine locale. I guess i can use Windows API SetLocaleInfo info to change the default.